### PR TITLE
Fix build scan command table printing with --vuln=true

### PR DESCRIPTION
**Pull request by dortam888**
- [ ] The pull request is targeting the `dev` branch.
- [ ] The code has been validated to compile successfully by running `go vet ./...`.
- [ ] The code has been formatted properly using `go fmt ./...`.
- [ ] All [static analysis checks](https://github.com/jfrog/jfrog-cli-security/actions/workflows/analysis.yml) passed.
- [ ] All [tests](https://github.com/jfrog/jfrog-cli-security/actions/workflows/test.yml) have passed. If this feature is not already covered by the tests, new tests have been added.
- [ ] All changes are detailed at the description. if not already covered at [JFrog Documentation](https://github.com/jfrog/documentation), new documentation have been added.

-----

Fixing issues when using --vuln=true flag and --format=table (default) in build scan. 
Issue 1: build scan printed vulnerability table twice instead of printing the violation table and the vulnerabilities table.
Issue 2: The vulnerability table is messed up, not sorted, and shows the same vulnerability multiple times.

### CODE REVIEW

File: commands/scan/buildscan.go [hunk 1]
```diff
@@ -155,7 +155,6 @@ func (bsc *BuildScanCommand) runBuildScanAndPrintResults(xrayManager *xray.XrayS
 
 	resultsPrinter := utils.NewResultsWriter(scanResults).
 		SetOutputFormat(bsc.outputFormat).
-		SetIncludeVulnerabilities(bsc.includeVulnerabilities).
 		SetIncludeLicenses(false).
```
Comment (hadarshjfrog) [inline commands/scan/buildscan.go:158]: after removing this line - did you test the CLI behaviour with and without watches?

Comment (dortam888) [reply commands/scan/buildscan.go:158]: @hadarshjfrog I did the regular tests (also with watches).
This line doesn't affect watches as it relates to vulnerabilities. (and watches are affecting violations).
We call SetIncludeVulnerabilities in line 179 if vulnerabilities should be added:
`		if bsc.includeVulnerabilities {
			resultsPrinter.**SetIncludeVulnerabilities**(true)
			if err = resultsPrinter.PrintScanResults(); err != nil {
				return false, err
			}
		}`

**Review by attiasas — CHANGES_REQUESTED** [review]
This change is not how to fix the issue. 
There are 2 issues here:
1. we can see that in the parsing of the flags, we don't take into account the `project` flag only in build-scan. the fix should add it, instead of removing and always printing violations...
image
image

3. In the table printing code, we do not expect to print both violations and vulnerabilities tables so we need to choose one (preferable vuln if requested)
image

**Comment by dortam888** [comment]
@attiasas @hadarshjfrog We have a couple of issues with jf bs.
Some parameters are missing, like projects and JAS enablement so we don't get results for builds from projects and we don't get JAS results at all (we don't have secrets but we can have CA).
Regarding this fix, See the comment in the code:
https://github.com/jfrog/jfrog-cli-security/blob/main/commands/scan/buildscan.go#L171
The behavior when the format is json for example is appending vulnerabilities to violations and that is what the code meant to do for tables as well based on the comment.
[Jira](https://jfrog-int.atlassian.net/browse/XRAY-72561) is about having violations and no vulnerabilities try to get vulnerabilites.
The build is failing because we have violations but we're not showing it in tables printing.

**Comment by attiasas** [comment]
fixed in:
https://github.com/jfrog/jfrog-cli-security/pull/163
https://github.com/jfrog/jfrog-cli-security/pull/164

**Pull request closed without merging**